_plugins/link-checker.rb

```diff
@@ -1,219 +1,283 @@
-# If set as `JEKYLL_FATAL_LINK_CHECKER=2`, the build will fail for internal and external dead links; in this case, there
-# is no need to set `JEKYLL_CHECK_EXTERNAL_LINKS`.
+# `JEKYLL_FATAL_LINK_CHECKER`, set on the environment, is the same as `JEKYLL_LINK_CHECKER`
+# except that it fails the build if there are broken links. it takes the same valid values
+# Usage: `JEKYLL_FATAL_LINK_CHECKER=internal bundle exec jekyll build --trace`
 
 module Jekyll::LinkChecker
-
+  ##
   # The collection that will get stores as the output
-  @urls = {}
 
+  @urls
+
+  ##
   # Pattern to identify documents that should be excluded based on their URL
-  @excluded_paths = /(\.(css|js|json|map|xml|txt|yml)$|\/version-selector\.tpl$)/i.freeze
 
+  @excluded_paths = %r{(\.(css|js|json|map|xml|txt|yml)$|/version-selector\.tpl$)}i.freeze
+
+  ##
   # Pattern to identify certain HTML tags whose content should be excluded from indexing
+
   @href_matcher = /<a[^>]+href=(['"])(.+?)\1/im.freeze
 
+  ##
   # Pattern to check for external URLs
-  @external_matcher = /^https?:\/\//.freeze
 
-  # List of domains to ignore
-  @ignored_domains = %w[localhost]
+  @external_matcher = %r{^https?://}.freeze
+  @forced_external_matcher = %r{^https?://.*(?=opensearch\.org/)}.freeze
 
+  ##
+  # List of domains to ignore
+  # playground.opensearch.org is causing an infinite redirect
+  # LinkedIn mostly fails with 999 status codes
+  @ignored_domains = [
+    'localhost',
+    'playground.opensearch.org', # inifite redirect, https://github.com/opensearch-project/dashboards-anywhere/issues/172
+    'crates.io', # 404s on bots
+    'www.cloudflare.com', # 403s on bots
+  ]
+
+  ##
   # Pattern of local paths to ignore
-  @ignored_paths = /(^\/javadocs\/)/.freeze
-
-  # Pattern to exclude when adding the `index.html` suffix to paths
-  @need_no_suffix = /\.(?!html)[^\/]+$/.freeze
-
-  # Valid response codes for successful links
-  @success_codes = %w[200 302]
-
-  # Questionable response codes for successful links
-  @questionable_codes = %w[301 403 429]
+  @ignored_paths = %r{(^/javadocs|^mailto:)}.freeze
 
+  ##
   # Holds the list of failures
-  @failures = []
-
-  # Driven by environment variables, it indicates a need to check external links
-  @check_external_links
-
-  # Driven by environment variables, it indicates the need to fail the build for dead links
-  @should_build_fatally
+  @failures
+
+  ##
+  # Build flags driven by environment variables
+  @@LINK_CHECKER_STATES = %w[internal forced all]
+  @check_links                # Enables the link checker
+  @check_forced_external      # Enables checking internal links marked as external e.g. /docs
+  @check_external_links       # Enables checking external links
+  @should_build_fatally       # indicates the need to fail the build for dead links
+
+  ##
+  # Defines the priority of the plugin
+  # The hooks are registered with a very low priority to make sure they runs after any content modifying hook
```

Comment:
"... make sure they run after ..." verb agreement
Not sure if it's necessary to make proper sentences above here and add periods. Just notes, I suppose.

Reply:
I won't change them all just to match the code in project-website repo for now.